catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -457,7 +510,18 @@ private void createListOfDeletedMetacards(List<Metacard> deletedMetacards,
                 LOGGER.info("Metacard creation exception creating metacards during delete", e);
                 throw new IngestException("Could not create metacard(s).");
             }
+        }
+    }
+
+    private void addPendingNrtDeletedMetacards(List<Metacard> deletedMetacards,
+            List<? extends Serializable> identifiers) {
+        if (deletedMetacards.size() < identifiers.size() && pendingNrtIndex.getAllPresent(
```

Comment:
As I'm understanding this, it is for the case where create/delete is called in rapid succession and the delete request needs to find the metacard to delete from the pending cache. I assume the request to delete will still be queued up by Solr and the NRT index will either
a) have the id added to it then quickly removed
b) never have the id added to it
?